_drafts/id-generation.md

```diff
@@ -0,0 +1,104 @@
+---
+layout: post
+title:  Distributed ID generation with Redis and Lua
+date:   2015-03-13 19:32:00
+excerpt: How to generate globally unique IDs in a distributed fashion with Redis and Lua.
+image: /assets/images/id-generation.jpg
+thumbnail: /assets/images/id-generation-thumbnail.jpg
+categories: engineering
+author: Nathan Kleyn
+---
+
+**TL;DR:** We are open-sourcing our distributed _k_-ordered ID generation project called "Icicle", which generates IDs using Lua scripting within distributed Redis hosts. You can [find more information in the GitHub repository](https://github.com/intenthq/icicle)!
+
+At Intent HQ, we generate a lot of data. Part of the task of capturing this data is the package it up into a common format and store it in a way it can be retrieved. Commonly, labelling this data in a way that facilitates retrival is a task handled by your database. However, in a big-data environment, we need a way to label these pieces of data with an ID in a distributed and scalable way independent of a database.
```

Comment:
`...capturing the data is the package it up...` -> `...capturing the data is TO package it up...`

`retrival` -> `retrieval`